src/tokens/saga.ts

```diff
@@ -309,6 +333,50 @@ export function* watchAccountFundedOrLiquidated() {
   }
 }
 
+export async function fetchImportedTokenBalances(
+  address: Address,
+  importedTokens: StoredTokenBalances,
+  knownTokenBalances: Record<string, FetchedTokenBalance>
+) {
+  try {
+    const importedTokensList = Object.values(importedTokens)
+    const importedTokensWithBalance: StoredTokenBalances = {}
+
+    const balanceRequests = importedTokensList.map(async (importedToken) => {
+      if (!importedToken) {
+        return
+      }
+
+      let balance
+      if (knownTokenBalances[importedToken.tokenId]) {
+        balance = knownTokenBalances[importedToken.tokenId].balance
```

Comment:
do we need to divide this by the token decimals?

Reply:
Yes! You are right!